docs/aws/audit/esmonitoring/rules/es_ri_recent_purchases_remediation.mdx

```diff
@@ -1,6 +1,251 @@
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent Elasticsearch Reserved Instance recent purchases from being misconfigured in AWS, you can follow these steps using the AWS CLI:
+
+1. **List All Reserved Instances:**
+   Use the following command to list all your Elasticsearch Reserved Instances. This helps you keep track of what you have and review recent purchases.
+
+   ```sh
+   aws es describe-reserved-elasticsearch-instances
+   ```
+
+2. **Monitor Reserved Instance Utilization:**
+   Regularly check the utilization of your Reserved Instances to ensure they are being used effectively. You can use CloudWatch metrics for this purpose.
+
+   ```sh
+   aws cloudwatch get-metric-statistics --namespace AWS/ES --metric-name ReservedInstanceUtilization --start-time <start-time> --end-time <end-time> --period 3600 --statistics Average
+   ```
+
+3. **Set Up Budget Alerts:**
+   Create budget alerts to notify you when spending on Reserved Instances exceeds a certain threshold. This helps in keeping track of unexpected purchases.
+
+   ```sh
+   aws budgets create-budget --account-id <account-id> --budget '{"BudgetName": "ElasticsearchReservedInstanceBudget", "BudgetLimit": {"Amount": "1000", "Unit": "USD"}, "TimeUnit": "MONTHLY", "BudgetType": "COST"}' --notifications-with-subscribers '[{"Notification": {"NotificationType": "ACTUAL", "ComparisonOperator": "GREATER_THAN", "Threshold": 80, "ThresholdType": "PERCENTAGE", "NotificationState": "ALARM"}, "Subscribers": [{"SubscriptionType": "EMAIL", "Address": "your-email@example.com"}]}]'
+   ```
+
+4. **Tagging Reserved Instances:**
+   Implement a tagging strategy for your Reserved Instances to easily identify and manage them. This can help in reviewing and auditing recent purchases.
+
+   ```sh
+   aws es add-tags --arn <reserved-instance-arn> --tag-list Key=Environment,Value=Production Key=Owner,Value=TeamA
+   ```
+
+By following these steps, you can effectively prevent misconfigurations related to Elasticsearch Reserved Instance recent purchases using the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Elasticsearch Reserved Instance recent purchases from being misconfigured, you can use Python scripts to automate the review and management of these instances. Below are the steps to achieve this:
+
+### 1. **Set Up AWS SDK for Python (Boto3)**
+First, you need to install and set up Boto3, the AWS SDK for Python, to interact with AWS services.
+
+```bash
+pip install boto3
+```
+
+### 2. **Authenticate and Initialize Boto3 Client**
+Create a Python script to authenticate and initialize the Boto3 client for Elasticsearch.
+
+```python
+import boto3
+
+# Initialize a session using Amazon Elasticsearch
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>